src/entry-server.js

```diff
@@ -26,23 +26,21 @@ export async function render({ req, res, buildManifest }) {
         return res.status(404).end();
     }
 
+    res.write(renderInitialBlock({ helmet: Helmet.renderStatic(), buildManifest }));
+
     // Render HTML that goes to into <div id="root"></div>
     const createElement = (Component, props) => <Component { ...props } serverContext={ { req, res } } />;
-    const rootHtml = renderToString(
+    const streamedRootHtml = renderToNodeStream(
         <RouterContext
             { ...renderProps }
             createElement={ createElement } />
     );
 
-    // Render document
-    const html = renderDocument({
-        helmet: Helmet.renderStatic(),
-        rootHtml,
-        buildManifest,
+    streamedRootHtml.pipe(res, { end: false });
```

Comment:
Need to investigate if we need to handle the error event. Usually the error event do not go through piping.